manifests/base/application-controller/argocd-application-controller-deployment.yaml

```diff
@@ -1,5 +1,5 @@
 apiVersion: apps/v1
-kind: StatefulSet
+kind: Deployment
```

Comment:
By `current sharding balance logic` do you mean the cluster distribution or the mechanism to compute the shard for each controller?

Reply:
The distribution algorithm. The default one depends on the statefulset to work irrc.

Reply:
Ok. After talking with @ishitasequeira it was clarified that the current distribution algorithm will work with deployment as it will also rely on the new configmap mechanism.